libpages/config/per_path_configs_v1.go

```diff
@@ -71,22 +83,48 @@ type accessControlV1 struct {
 	// with `list`, causing maxPermission to be {read: true, list: true} but
 	// never a user getting both.
 	maxPermission permissionsV1
-	// p stores the path (from Config declaration) that an *accessControlV1
-	// object is constructed for. When an *aclCheckerV1 is picked for a path,
+	// p stores the path (from Config declaration) that an *perPathConfigV1
+	// object is constructed for. When an *perPathConfigsReaderV1 is picked for a path,
 	// the p field can be used as a realm for HTTP Basic Authentication.
 	p string
+
+	accessControlAllowOrigin string
+	custom403Forbidden       string
+	custom404NotFound        string
+}
+
+func checkCors(acao string) (cleaned string, err error) {
+	cleaned = strings.TrimSpace(acao)
+	if cleaned != "" && cleaned != "*" {
+		// TODO: support setting non-wildcard origins. Note that none wildcard
+		// ones need a Vary header too.
+		return "", ErrInvalidConfig{msg: "only \"*\" is supported as " +
+			"non-empty Access-Control-Allow-Origin for now"}
+	}
+	return cleaned, nil
+}
+
+func checkCustomPagePath(p string) (cleaned string, err error) {
+	if len(p) == 0 {
+		return "", nil
+	}
+	cleaned = path.Clean(p)
+	if strings.HasPrefix(cleaned, "..") || !strings.HasSuffix(cleaned, ".html") {
```

Comment:
Is the suffix restriction needed?  Why not allow `.txt` suffixes, or other static file types?